tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -300,6 +363,29 @@ public:
    static std::unique_ptr<RNTupleProcessor> CreateJoin(const std::vector<RNTupleOpenSpec> &ntuples,
                                                        const std::vector<std::string> &joinFields,
                                                        std::vector<std::unique_ptr<RNTupleModel>> models = {});
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Create a new RNTuple processor for horizontally combined RNTuples.
+   ///
+   /// \param[in] ntuples A list specifying the names and locations of the ntuples to process. The first ntuple in the
+   /// list will be considered the primary ntuple and drives the processor iteration loop. Subsequent ntuples are
+   /// considered auxiliary, whose entries to be read are determined by the primary ntuple (which does not necessarily
+   /// have to be sequential).
+   /// \param[in] joinFields The names of the fields on which to join, in case the specified ntuples are unaligned.
+   /// The join is made based on the combined join field values, and therefore each field has to be present in each
+   /// specified RNTuple. If an empty list is provided, it is assumed that the specified ntuple are fully aligned, and
+   /// `RNTupleIndex` will not be used.
+   /// \param[in] processorName The name to give to the processor. Use
+   /// CreateJoin(const std::vector<RNTupleOpenSpec> &, const std::vector<std::string> &, std::unique_ptr<RNTupleModel>)
+   /// to automatically use the name of the input RNTuple instead.
+   /// \param[in] models A list of models for the ntuples. This list must either contain a model for each ntuple in
+   /// `ntuples` (following the specification order), or be empty. When the list is empty, the default model (i.e.
+   //// Scontaining all fields) will be used for each ntuple.
+   ///
+   /// \return A pointer to the newly created RNTupleProcessor.
+   static std::unique_ptr<RNTupleProcessor>
+   CreateJoin(const std::vector<RNTupleOpenSpec> &ntuples, const std::vector<std::string> &joinFields,
+              std::string_view processorName, std::vector<std::unique_ptr<RNTupleModel>> models = {});
```

Comment:
Looking again at this factory, one possibility to make the interface more prescriptive could be having two overloads, this one could just become
```suggestion
   static std::unique_ptr<RNTupleProcessor>
   CreateJoin(const std::vector<RNTupleOpenSpec> &ntuples, const std::vector<std::string> &joinFields,
              std::string_view processorName);
```
which would only use the default model, then we could have a new one
```cpp
   static std::unique_ptr<RNTupleProcessor>
   CreateJoin(
  const std::vector<std::pair<RNTupleOpenSpec, std::unique_ptr<RNTupleModel>> &ntuplesWithModels,
  const std::vector<std::string> &joinFields,
  std::string_view processorName);
```